pipelinewise/fastsync/commons/transform_utils.py

```diff
@@ -330,3 +350,36 @@ def __mask_date_to_sql(cls, column: str, sql_flavor: SQLFlavor) -> str:
             )
 
         return trans
+
+    @classmethod
+    # pylint: disable=W0238  # False positive when it is used by another classmethod
+    def __mask_string_skip_ends_to_sql(
+        cls, transform_type: TransformationType, column: str, sql_flavor: SQLFlavor
+    ) -> str:
+        """
+        convert MASK-STRING-SKIP-ENDS-n transformation into the right sql string
+        Args:
+            column: column to apply the masking to
+            sql_flavor: the sql flavor to use
+
+        Raises: NotImplementedError if mask-string-skip-ends is not implemented for the given sql flavor
+
+        Returns: sql string equivalent of the mask-string-skip-ends
+        """
+        skip_ends_n = int(transform_type.value[-1])
+
+        if sql_flavor == SQLFlavor.SNOWFLAKE:
+            trans = '{0} = CASE WHEN LENGTH({0}) > 2 * {1} THEN ' \
+                    'CONCAT(SUBSTRING({0}, 1, {1}), REPEAT(\'*\', LENGTH({0})-(2 * {1})), ' \
+                    'SUBSTRING({0}, LENGTH({0})-{1}+1, {1})) ' \
+                    'ELSE {0} END'.format(column, skip_ends_n)
+        elif sql_flavor == SQLFlavor.POSTGRES:
+            trans = '{0} = CASE WHEN LENGTH({0}) > 2 * {1} THEN ' \
```

Comment:
Same here, I believe that this is not masking the entire value if it's not long enough, which was something we discussed and changed in the transform-field.

Reply:
Done